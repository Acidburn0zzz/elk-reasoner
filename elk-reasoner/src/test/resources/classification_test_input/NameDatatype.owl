Prefix(:=<http://www.semanticweb.org/Test.owl#>)
Prefix(owl:=<http://www.w3.org/2002/07/owl#>)
Prefix(rdf:=<http://www.w3.org/1999/02/22-rdf-syntax-ns#>)
Prefix(xml:=<http://www.w3.org/XML/1998/namespace>)
Prefix(xsd:=<http://www.w3.org/2001/XMLSchema#>)
Prefix(rdfs:=<http://www.w3.org/2000/01/rdf-schema#>)

Ontology(<http://www.semanticweb.org/Test.owl>

EquivalentClasses(:_valid_string_A	DataHasValue(:dp "This is a string!"))
EquivalentClasses(:_valid_string_B	DataHasValue(:dp "Édition française."))
EquivalentClasses(:_valid_string_C	DataHasValue(:dp "AT&amp;T"))
EquivalentClasses(:_valid_string_D	DataHasValue(:dp "12.5"))
EquivalentClasses(:_valid_string_E	DataHasValue(:dp ""))
EquivalentClasses(:_valid_string_F	DataHasValue(:dp "   Отступ 3 пробела   "))
EquivalentClasses(:_valid_string_G	DataHasValue(:dp " Отступ 1 пробел "))
EquivalentClasses(:_valid_string_H	DataHasValue(:dp "	Tabulated text	"))
EquivalentClasses(:_valid_string_I	DataHasValue(:dp ":Foo."))
EquivalentClasses(:_valid_string_J	DataHasValue(:dp "_Foo."))
EquivalentClasses(:_valid_string_K	DataHasValue(:dp "Multiline 
									text"))

EquivalentClasses(:AllStrings		DataSomeValuesFrom(:dp xsd:string))
EquivalentClasses(:AllNames		DataSomeValuesFrom(:dp xsd:Name))
EquivalentClasses(:AnyNonEmptyName	DataSomeValuesFrom(:dp DatatypeRestriction(xsd:Name xsd:minLength "1"^^xsd:integer)))
EquivalentClasses(:AnyShortName		DataSomeValuesFrom(:dp DatatypeRestriction(xsd:Name xsd:minLength "0"^^xsd:integer xsd:maxLength "5"^^xsd:integer)))
EquivalentClasses(:NumericName		DataSomeValuesFrom(:dp DatatypeRestriction(xsd:Name xsd:pattern "[0-9., ]+")))

EquivalentClasses(:InvalidNameA	DataSomeValuesFrom(:dp DatatypeRestriction(xsd:Name xsd:maxLength "5"^^xsd:integer xsd:minLength "10"^^xsd:integer)))
EquivalentClasses(:InvalidNameB	DataSomeValuesFrom(:dp DatatypeRestriction(xsd:Name xsd:maxLength "-2"^^xsd:integer)))
EquivalentClasses(:InvalidNameC	DataSomeValuesFrom(:dp DatatypeRestriction(xsd:Name xsd:minLength "-10"^^xsd:integer)))
)