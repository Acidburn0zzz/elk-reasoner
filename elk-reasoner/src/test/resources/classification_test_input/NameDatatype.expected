Prefix(xsd:=<http://www.w3.org/2001/XMLSchema#>)
Prefix(owl:=<http://www.w3.org/2002/07/owl#>)
Prefix(:=<http://www.semanticweb.org/Test.owl#>)
Prefix(xml:=<http://www.w3.org/XML/1998/namespace>)
Prefix(rdf:=<http://www.w3.org/1999/02/22-rdf-syntax-ns#>)
Prefix(rdfs:=<http://www.w3.org/2000/01/rdf-schema#>)

Ontology(<http://www.semanticweb.org/Test.owl>

SubClassOf(:AllNames		:AllStrings)
SubClassOf(:AnyNonEmptyName	:AllNames)
SubClassOf(:AnyShortName	:AllNames)

SubClassOf(:_valid_string_A	:AllStrings)
SubClassOf(:_valid_string_B	:AllStrings)
SubClassOf(:_valid_string_C	:AllStrings)
SubClassOf(:_valid_string_D	:AllStrings)
SubClassOf(:_valid_string_E	:AllStrings)
SubClassOf(:_valid_string_F	:AllStrings)
SubClassOf(:_valid_string_G	:AllStrings)
SubClassOf(:_valid_string_H	:AllStrings)
SubClassOf(:_valid_string_K	:AllStrings)
SubClassOf(:_valid_string_I	:AnyNonEmptyName)
SubClassOf(:_valid_string_I	:AnyShortName)
SubClassOf(:_valid_string_J	:AnyNonEmptyName)
SubClassOf(:_valid_string_J	:AnyShortName)

EquivalentClasses(:InvalidNameA :InvalidNameB :InvalidNameC :NumericName owl:Nothing)
)
